internal/modsdir/manifest.go

```diff
@@ -48,7 +49,11 @@ type Record struct {
 type Manifest map[string]Record
 
 func (m Manifest) ModuleKey(path addrs.Module) string {
-	return path.String()
```

Comment:
This change allows `init` to continue to work, while we get a more human-understandable `String()` representation of `Module` -- for example, this will result in `child` vs `module.child`